src/main/java/io/vertx/core/file/impl/FileResolverImpl.java

```diff
@@ -294,18 +296,25 @@ private File unpackFromJarURL(URL url, String fileName, ClassLoader cl) {
         File file = resolveFile(s);
         zip = new ZipFile(file);
       }
-
+      String inJarPath = path.substring(idx1 + 6);
+      String[] parts = JAR_URL_SEP_PATTERN.split(inJarPath);
```

Comment:
rather use a Matcher for the pattern and build the prefix using matcher index and append(String, int, int)

Reply:
Since we just need to split the path with `!/`, I would like use `String.indexOf(String, int)`, and leave Pattern away.